### PR TITLE
Multiple send detector

### PR DESCRIPTION
**Pull request by yan**
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/1084)

### CODE REVIEW

File: manticore/ethereum.py [hunk 1]
```diff
@@ -631,6 +631,26 @@ def did_evm_write_storage_callback(self, state, address, offset, value):
         state.context.setdefault('{:s}.initialized_storage'.format(self.name), set()).add((address, offset))
 
 
+class DetectMultipleSends(Detector):
+    '''
+    Detects multiple sends to the same address within a transaction
+    '''
+    def will_evm_execute_instruction_callback(self, state, instruction, arguments):
```
Comment (feliam) [inline manticore/ethereum.py]: Should we do this after the fact and only if the transaction was successful?
Also not sure if it is important for the current goal but you can send value with CREATE, and SELFDESTRUCT :) and DELEGATECALL 
all open_transactions + delegatecall

File: manticore/ethereum.py [hunk 2]
```diff
@@ -631,6 +631,26 @@ def did_evm_write_storage_callback(self, state, address, offset, value):
         state.context.setdefault('{:s}.initialized_storage'.format(self.name), set()).add((address, offset))
 
 
+class DetectMultipleSends(Detector):
+    '''
+    Detects multiple sends to the same address within a transaction
```
Comment (mossberg) [inline manticore/ethereum.py]: I actually don't think that it's particularly a problem if the transactions are to the same address. My understanding is that it's just a general not-great practice to have more than one transaction initiated (.send, .transfer, .call, ...) within a function

**Comment by mossberg** [comment]
I am a little mixed on if we should ultimately include this detector 🤔 . Our detectors can be seen as detecting findings on spectrum from "maybe bug" to "just a code quality issue", and i think this falls very far on the code quality issue side of things, maybe almost to the point where we don't need it. i don't really think manticore is the tool for finding these kinds of code quality issues, i think slither would probably be a better fit, anyway?

**Comment by mossberg** [comment]
confirmed that slither can provide these capabilities if desired. also confirmed that we probably don't want to warn on this general multiple sends scenario in all cases, since it is not directly representative of a finding or issue. going to close

**Pull request closed without merging**